### PR TITLE
[MRG] Fix threshold must be monotonic condition

### LINKED ISSUE #4724: confliciting logic?
Hey there,

I wanted to see with you if I'm missing something when running a spatio_temporal_cluster_test with tail=-1 and tfce thresholds. 

It seems that with tail=-1, one must use negative start and step tfce values:
https://github.com/mne-tools/mne-python/blob/966e862b95861de47ffe259bd66c86feece524f6/mne/stats/cluster_level.py#L301-L306

When doing so, the check if thresholds are monotonic fails.
https://github.com/mne-tools/mne-python/blob/966e862b95861de47ffe259bd66c86feece524f6/mne/stats/cluster_level.py#L335-L336
Should this check be for absolute values?

### PR DESCRIPTION
**Pull request by cbrnr**
Fix #4724

### CODE REVIEW

**Comment by agramfort** [comment]
test?

**Comment by larsoner** [comment]
I'm going to code up a Gramfort-CI that looks at the `diff` and if there are no `test_` files changed, writes:
```
test?
```

**Comment by cbrnr** [comment]
🤣 

BTW, the test I've just added does not test for the reported issue, but only for the already existing checks:

https://github.com/cbrnr/mne-python/blob/230d097f3528f83b96d4447482d8867a91aff7a7/mne/stats/cluster_level.py#L301-L306

How do I add a test that checks if no error is raised? The following call previously caused an error:

    permutation_cluster_test(data, tail=-1, threshold=dict(start=0, step=-0.1))

But now works with this fix.

**Comment by larsoner** [comment]
You can call the function but don't need to check any of its outputs (smoke test)

**Comment by cbrnr** [comment]
OK good, so I'll add this test as well. Furthermore, I have no idea how to trigger the "Thresholds must be monotonically increasing" errors I just changed - ideas?

**Comment by larsoner** [comment]
`tail=1, threshold=dict(start=1, step=-0.5)` and `tail=-1, threshold=dict(start=-1, step=0.5)` should do it

**Comment by cbrnr** [comment]
Never mind, this is just a warning... OK to use a test that generates a warning?

**Comment by larsoner** [comment]
We don't want un-caught warnings, you can catch it with `warnings.catch_warnings(record=True):` context manager or just use `randn` instead of `rand`

**Comment by larsoner** [comment]
(and remove the subtraction, or multiply your existing values by 10, etc.)

**Comment by cbrnr** [comment]
I can't get it to work by increasing numbers etc., these are the max values of the permutation statistics and not the input data. The above tests catch the warnings as you suggest, so I'll do the same.

**Comment by cbrnr** [comment]
`tail=-1, threshold=dict(start=-1, step=0.5)` triggers the first error (because `step` must be negative for `tail==-1`).

**Comment by cbrnr** [comment]
Also, how can I catch a specific `ValueError` with `assert_raises`?

**Comment by larsoner** [comment]
We haven't so far -- there is `assert_raises_regex` in `numpy` I think (but might be too new NumPy for us), `pytest` might also provide something

**Comment by cbrnr** [comment]
OK, then let's better not use it for now. Can you think of a way to trigger the monotonic error for `tail=-1`?

**Comment by larsoner** [comment]
No, it sounds like the checks might be redundant, then

**Comment by cbrnr** [comment]
Cool, let's see if Travis, and more importantly @agramfort is happy now :smile:

**Comment by larsoner** [comment]
If you want, fine with me either way

**Comment by cbrnr** [comment]
After playing around with this for a while I give up. No matter how I change the data I always get the warning. For this test, I'd really like to avoid this warning though.

**Comment by larsoner** [comment]
Maybe there is something wrong with the warning, then?

**Comment by cbrnr** [comment]
Yes, hmmmm. The problem is that it's using the F statistic, which is always positive. Then it constructs the thresholds array as follows:

    thresholds = np.arange(threshold['start'], stop, threshold['step'], float)

Here, `threshold['step']` is negative (because `tail=-1`) and `stop` is positive. Therefore, `thresholds` is empty and hence the warning.

**Comment by larsoner** [comment]
So for this function using negative values actually doesn't make any sense. You can use the 1-sample tests (which use t) to avoid this

**Comment by agramfort** [comment]
don't forget to update what's new

**Comment by cbrnr** [comment]
Alright, this should be it. Ready to merge if CIs return happy.

File: doc/whats_new.rst [hunk 1]
```diff
@@ -29,6 +29,7 @@ Bug
 ~~~
 
 - Fix bug in :meth:`mne.io.set_eeg_reference` to remove an average reference projector when setting the reference to ``[]`` (i.e. do not change the existing reference) by `Clemens Brunner`_
```
Comment (agramfort) [inline doc/whats_new.rst]: missing empty line

**Comment by cbrnr** [comment]
I don't know how you normally handle PRs, but in this case it would be good to squash all commits into one...

**Comment by agramfort** [comment]
it gets squashed on merge.

**Comment by cbrnr** [comment]
Except for the OSX jobs all CIs are happy.

**Comment by larsoner** [comment]
Thanks @cbrnr

**Pull request merged**